x/dex/keeper/liquidity.go

```diff
@@ -43,20 +42,6 @@ func (k Keeper) Swap(
 
 		inAmount, outAmount := liq.Swap(remainingTakerDenom, remainingMakerDenom)
 
-		// If (due to rounding) the actual price given to the maker is demonstrably unfair
-		// we do not save the results of the swap and we exit.
-		// While the decrease in price quality for the maker is semi-linear with the amount
-		// being swapped, it is possible that the next swap could yield a "fair" price.
-		// Nonetheless, once the remainingTakerDenom gets small enough to start causing unfair swaps
-		// it is much simpler to just abort.
-		if inAmount.IsZero() || isUnfairTruePrice(params.MaxTrueTakerSpread, inAmount, outAmount, liq) {
```

Comment:
now that we removed this check - i see you've added the check for truePrice < exitLimitPrice.
Do we have any tests for cases where this condition is not satisfied?

Reply:
https://github.com/neutron-org/neutron/blob/fix%2Frounding_changes/x/dex/keeper/integration_placelimitorder_test.go#L471
https://github.com/neutron-org/neutron/blob/fix%2Frounding_changes/x/dex/keeper/integration_placelimitorder_test.go#L304